Update to buienradar json api; and additional monitored_conditions

mjj4791 opened the pull request:
## Breaking Change:
The following sensor types (monitored_conditions) are no longer supported, since they are no longer provided by the json api of buienradar:

- snow_1d .. snow_5d

The following monitored conditions will change unit:

- windgust (now km/h, was m/s)
- windspeed  (now km/h, was m/s)
- windspeed_?d  (now km/h, was m/s)
- visibility (now km, was m)

## Description:
Switching over to new version of python-buienrader (1.0.1); this version now leverages the new json buienradar-api. This API provides new sensor data to home-assistant:

- barometerfc
- barometerfcname
- barometerfcnamenl
- feeltemperature
- rainlast24hour
- rainlasthour
- minrain_1d .. minrain_5d
- maxrain_1d .. maxrain_5d

Forecast data (Weather component) will now also contain:
- windbearing
- windspeed

**Related issue (if applicable):** 
Pyhton-buienradar 1.0.1 now also includes the buienradar radar url functionality, requested here: https://github.com/home-assistant/home-assistant/pull/23358

unit conversion from m and m/s to km and km/h, as mentioned here:
https://github.com/home-assistant/home-assistant/issues/20142
https://github.com/home-assistant/home-assistant/issues/18558

**Pull request with documentation for [home-assistant.io](https://github.com/home-assistant/home-assistant.io) (if applicable):** home-assistant/home-assistant.io#9605


## Checklist:
  - [X] The code change is tested and works locally.
  - [X] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [X] There is no commented out code in this PR.
  - [X] I have followed the [development checklist][dev-checklist]

If user exposed functionality or configuration variables are added/changed:
  - [X] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [X] [_The manifest file_][manifest-docs] has all fields filled out correctly. Update and include derived files by running `python3 -m script.hassfest`.
  - [X] New or updated dependencies have been added to `requirements_all.txt` by running `python3 -m script.gen_requirements_all`.
  - [X] Untested files have been added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Tests have been added to verify that the new code works.

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html

mjj4791 committed:
Update to json api; and additional sensors

mjj4791 committed:
remove unneeded coordinate rounding

OttoWinter commented on homeassistant/components/buienradar/weather.py:
```diff
@@ -150,25 +152,33 @@ def temperature_unit(self):
+                                          WINDSPEED)
+
+        if not self._forecast:
+            return
```
```suggestion
            return None
```

Explicit None return if other returns return a value.

OttoWinter commented on homeassistant/components/buienradar/weather.py:
```diff
@@ -150,25 +152,33 @@ def temperature_unit(self):
+            data_out = {}
+            condcode = data_in.get(CONDITION, []).get(CONDCODE)
+
+            data_out[ATTR_FORECAST_TIME] = data_in.get(DATETIME)
```
Could be simplified to
```python
data_out = {
  ATTR_FORECAST_TIME: data_in.get(DATETIME),
  ATTR_FORECAST_CONDITION: cond[condcode],
  # ...
}
```

mjj4791 committed:
minor optimizations

mjj4791 commented:
I processed the requested review items; not sure what I can do about the conflicting files?

mjj4791 committed:
Update CODEOWNERS

mjj4791 committed:
Update sensor.py

unit conversion for:
- windgust (ms to km/h)
- windspeed (m/s to km/h)
- windspeed_1d (ms/km/h)
- visibility (m to km)

mjj4791 committed:
Update weather.py

unit conversion for windspeed (m/s to km/h)

mjj4791 committed:
Update sensor.py

mjj4791 committed:
Update weather.py

mjj4791 committed:
Update weather.py

mjj4791 committed:
Update CODEOWNERS

mjj4791 committed:
Update manifest.json

mjj4791 committed:
Update CODEOWNERS

mjj4791 commented:
@OttoWinter 
Can you be of assistance resolving the conflicts in manifest/codeowners?

Seems we have 2 owners in one component now (@mjj4791 for sensor/weather and @Ties for the camera component in buienradar). However I am unable to convince my update/commit to manifest and codeowners to get accepted...

ties commented:
@mjj4791 I think this is similar to what happened when Paulus added me as as a codeowner in the manifest. I think that the linter that checks that that `CODEOWNERS` is identical to the output generated by the script from the manifest metadata.

I think that is generated [by](https://developers.home-assistant.io/docs/en/development_checklist.html) `python3 -m script.hassfest`

mjj4791 committed:
Update CODEOWNERS

yet another try to get CODEOWNERS accepted...

mjj4791 commented:
I ran `python3 -m script.hassfest ` and updated CODEOWNERS to match; 
still no luck getting the file to be accepted....

ties commented:
It seems to pass the checks but it has merge errors. I rebased the branch from dev, skipping empty commits. Doing that and force pushing should make this mergeable.

My result is in https://github.com/ties/home-assistant/tree/feature/pr24463_rebased

MartinHjelmare commented:
Please rebase on latest dev branch and solve the merge conflicts.

MartinHjelmare reviewed (approved):
Can be merged when build passes.

MartinHjelmare commented:
@mjj4791 please rebase so we can merge this.

mjj4791 committed:
update codeowners

homeassistant commented:
Hello @mjj4791,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`25fde40b958e63c6dc8643de4014e4769d6a527f`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant&#x2F;pull&#x2F;24463&#x2F;commits&#x2F;25fde40b958e63c6dc8643de4014e4769d6a527f) - This commit has something that looks like an email address (mj041422@cerner.net). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

mjj4791 committed:
update codeowners

homeassistant commented:
Hello @mjj4791,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`28129981429b294783b0676e25e9fa6fa315b783`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant&#x2F;pull&#x2F;24463&#x2F;commits&#x2F;28129981429b294783b0676e25e9fa6fa315b783) - This commit has something that looks like an email address (mj041422@cerner.net). Maybe try linking that to GitHub?.
- [`25fde40b958e63c6dc8643de4014e4769d6a527f`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant&#x2F;pull&#x2F;24463&#x2F;commits&#x2F;25fde40b958e63c6dc8643de4014e4769d6a527f) - This commit has something that looks like an email address (mj041422@cerner.net). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

mjj4791 committed:
update

homeassistant commented:
Hello @mjj4791,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`8ac8ad9748eb93c2c074e50bf2ef94136ba5557d`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant&#x2F;pull&#x2F;24463&#x2F;commits&#x2F;8ac8ad9748eb93c2c074e50bf2ef94136ba5557d) - This commit has something that looks like an email address (mj041422@cerner.net). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

MartinHjelmare commented:
File CODEOWNERS is not up to date. Please run python3 -m script.hassfest

mjj4791 committed:
updated codeowners to match manifest
